### DISCUSSION TITLE
What is the best way to update?

### DISCUSSION THREAD

**Discussion by LamerManuel** [satnaing/astro-paper#31] [Q&A]
Tried to update it via "npm update" but unfortunately that doesn't work. Therefore my question, what is the best way to update the theme? Couldn't find anything in the readme.

**Comment by satnaing** [comment]
Hello @LamerManuel 
What do you want to update? AstroPaper 1 to AstroPaper v2? Or want to update third-party packages? For third-party packages I usually use [npm-check-updates](https://www.npmjs.com/package/npm-check-updates) package.

**Reply by elicore** [reply]
How do I update v2 to the latest version? Thanks!

**Reply by bytrangle** [reply]
Hi @satnaing . You have refactored this theme a lot since the last time I used them. Is there any way to update those changes in my repo? When running `npm create astro@latest --template satnaing/astro-paper`, are users supposed to get a fixed version of Astro Paper at the time?

**Comment by bowencool** [comment]
Add a new `remote` to your git repo:

```sh
git remote add upstream https://github.com/satnaing/astro-paper.git
# check
git remote -v
```
and you will be able to merge branch `upstream/main` into yours or rebase your branch onto `upstream/main`.

**Reply by bytrangle** [reply]
Thank you.

**Comment by k4u5hik** [comment]
The solution should be in the README. I came here seeking an answer to the same question.

**Comment by eerison** [comment]
Duplicate: https://github.com/satnaing/astro-paper/discussions/393

**Comment by eerison** [comment]
Hello guys :)

just to share with you that I created 2 articles with alternatives how to update your blog.

an simple one: https://blog.erison.work/posts/get-frontend-theme-updates-using-git-part-1/
and another one keeping the git history clean: https://blog.erison.work/posts/get-frontend-theme-updates-using-git-part-2/

I split into 2 articles, because it gonna be so big.

I hope it helps you update your projects, and in case you face any issue, let me know in the comments :)
